plugin/src/main/java/org/apache/hertzbeat/plugin/Plugin.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.hertzbeat.plugin;
+
+import org.apache.hertzbeat.common.entity.alerter.Alert;
+
+/**
+ * Plugin
+ */
+public interface Plugin {
```

Comment:
Is this interface just for custom alerting? If so, it is recommended that the name be changed to `AlertPlugin`.

Reply:
Just a preliminary implementation, the current timing is only after the alarm, the follow-up can be added to an enumeration class to specify the timing, such as the collection, triggering the alarm silence, triggering the alarm convergence, etc.

Reply:
What I mean is that it is best to define the interface at the beginning, otherwise subsequent changes may involve break changes, leading to compatibility issues.

```java
public interface Plugin {
    void execute(Alert alert);
}
```

The interface's name is `Plugin`, and the method is `execute(Alert alert)`.

They are all common names but they only do one thing: alerting. Maybe they can be more focused, such as:

```java
public interface AlertPlugin {
     void execute(Alert alert);
}

// or
public interface Plugin {
     void alert(Alert alert);
}
```